src/OrchardCore/OrchardCore.ResourceManagement/TagHelpers/StyleTagHelper.cs

```diff
@@ -241,12 +244,53 @@ public override async Task ProcessAsync(TagHelperContext context, TagHelperOutpu
                 output.Content.AppendHtml(sw.ToString());
             }
         }
-        else if (string.IsNullOrEmpty(Name) && string.IsNullOrEmpty(Src))
+        else
         {
             // Custom style content
 
             var childContent = await output.GetChildContentAsync();
 
+            if (!string.IsNullOrEmpty(DependsOn))
+            {
+                var dependencies = DependsOn.Split(',', StringSplitOptions.TrimEntries | StringSplitOptions.RemoveEmptyEntries);
+
+                foreach (var dependency in dependencies)
+                {
+                    var dependencyParts = dependency.Split(';');
```

Comment:
Location is not applicable to styles.

Reply:
Does this not work?

<style at="Foot">

</style>

Reply:
It'll display the code, but it'll do this in the `<head>`, just as `at="Head"`. It should actually also do this if you don't specify a location too (since unlike `<script>`, a `<style>` tag is not valid in `body`), and while `<style asp-src="..."></style>` does this, inline style doesn't. So, that's a bug, and it should work the same with any location, and not require one.

Reply:
Please check the updated code.